apis/v1alpha2/shared_types.go

```diff
@@ -249,4 +238,90 @@ type RouteStatus struct {
 //
 // +kubebuilder:validation:MinLength=1
 // +kubebuilder:validation:MaxLength=253
+// +kubebuilder:validation:Pattern=`^(\*\.)?[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
 type Hostname string
+
+// Group refers to a Kubernetes Group. It must either be an empty string or a
+// RFC 1123 subdomain.
+//
+// This validation is based off of the corresponding Kubernetes validation:
+// https://git.k8s.io/apimachinery/pkg/util/validation/validation.go#L208
+//
+// Valid values include:
+//
+// * ""
+// * example.com
+// * foo.example.com
+//
+// Invalid values include:
+//
+// * example.com/bar
+//
+// +kubebuilder:validation:MaxLength=253
+// +kubebuilder:validation:Pattern=`^$|^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
+type Group string
+
+// Kind refers to a Kubernetes Kind.
+//
+// +kubebuilder:validation:MinLength=1
+// +kubebuilder:validation:MaxLength=63
+// +kubebuilder:validation:Pattern=`^[a-zA-Z]([-a-zA-Z0-9]*[a-zA-Z0-9])?$`
+type Kind string
+
+// Namespace refers to a Kubernetes namespace. It must be a RFC 1123 label.
+//
+// This validation is based off of the corresponding Kubernetes validation:
+// https://git.k8s.io/apimachinery/pkg/util/validation/validation.go#L187
+//
+// This is used for Namespace name validation here:
+// https://git.k8s.io/apimachinery/pkg/api/validation/generic.go#L63
```

Comment:
These line references will get out of date. Opt for only the file or even better, target a specific commit like so:
https://github.com/kubernetes/apimachinery/blob/02cfb53916346d085a6c6c7c66f882e3c6b0eca6/pkg/api/validation/generic.go#L63

Open the above link and press 'y'.

Reply:
Updated, thanks! Thanks for letting me know about that "y" shortcut, super helpful.